### PR TITLE
Cleanup BindingManagerBase

### PR DESCRIPTION
**Pull request by hughbe**
Already tested :)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/BindingManagerBase.cs [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+
+        protected internal abstract string GetListName(ArrayList listAccessors);
+
+        public abstract void SuspendBinding();
+
+        public abstract void ResumeBinding();
+
+        protected void PullData()
+        {
+            bool success;
```
Comment (JuditRose) [inline src/System.Windows.Forms/src/System/Windows/Forms/BindingManagerBase.cs]: Is there a reason not to clean up the variable success here as you did in the method PushData?

Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/BindingManagerBase.cs]: Fixed!

File: src/System.Windows.Forms/src/System/Windows/Forms/CurrencyManager.cs [hunk 2]
```diff
@@ -815,8 +815,8 @@ public class CurrencyManager : BindingManagerBase {
         // and when the index of the ListChangedEventArgs == the position in the currency manager
         /// <include file='doc\ListManager.uex' path='docs/doc[@for="CurrencyManager.OnCurrentItemChanged"]/*' />
         internal protected override void OnCurrentItemChanged(EventArgs e) {
-            if (onCurrentItemChangedHandler != null) {
-                onCurrentItemChangedHandler(this, e);
+            if (_onCurrentItemChangedHandler != null) {
+                _onCurrentItemChangedHandler(this, e);
```
[the remark this answers is not in the document]
Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/CurrencyManager.cs]: Fixed

File: src/System.Windows.Forms/src/System/Windows/Forms/BindingManagerBase.cs [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+                    {
+                        bi.BindingComplete += new BindingCompleteEventHandler(Binding_BindingComplete);
+                    }
+                    break;
+            }
+        }
+
+        private void OnBindingsCollectionChanging(object sender, CollectionChangeEventArgs e)
+        {
+            if (e.Action == CollectionChangeAction.Refresh)
```
[the remark this answers is not in the document]
Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/BindingManagerBase.cs]: fixed

**Pull request merged**